src/streamlink/plugins/showroom.py

```diff
@@ -78,6 +79,46 @@
 ))
 
 
+# base reload time on segment duration rather than TARGETDURATION,
+# as the latter is consistently wrong
+class ShowroomHLSStreamWorker(hls.HLSStreamWorker):
+    def process_sequences(self, playlist, sequences):
```

Comment:
You should not rewrite the whole `process_sequences` method here. Once the parent class gets changed, your custom one can break. The `HLSStreamWorker` class is not that flexible unfortunately and needs a bit of refactoring, but for now you could move the reload time caluclation into a separate method and then just override that in your custom class, like it's been done in my Twitch low latency PR:
https://github.com/streamlink/streamlink/pull/2513/files#diff-7ddabf15fb3a46558218929a060133e0R225-R237

Reply:
Right... but is that going to be merged soon? I didn't want to touch hls.py

Reply:
> but is that going to be merged soon?

The linked PR probably not, but that's why I meant that you should add the same change in your PR.

It's a reasonable change and there's no reason why your PR shouldn't get merged if you improve the basic HLS class with that necessary change. If you want to keep this PR as simple as possible, open another one with the HLS class change alone and rebase this one after the other has been merged.